File: source-code/onboarding/core/src/main/java/bo/com/mc4/onboarding/core/service/business/CanalOnboardingService.java
package bo.com.mc4.onboarding.core.service.business;

import bo.com.mc4.onboarding.model.business.dto.CanalOnboardingDto;
import bo.com.mc4.onboarding.model.business.dto.ConsultoraRecomendanteDto;
import bo.com.mc4.onboarding.model.business.dto.DirectoraDto;

import java.util.List;

public interface CanalOnboardingService {
    List<CanalOnboardingDto> canalesOnboardingList();
}

File: source-code/onboarding/backoffice-web/src/main/web/src/app/commons/services/business/canal-onboarding.service.ts
import { HttpClient, HttpParams } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { map } from "rxjs/operators";
import { Observable } from "rxjs";
import { environment } from "src/environments/environment";
import { ApiResponse } from "../../model/api-response";
import { IEnumDto } from "../../model/enum.value";
import { mapResponseApi } from "../../utils/http.util";
import { CanalOnboardingDto } from "../../model/business.interface";

@Injectable({
  providedIn: "root",
})
export class CanalOnboardingService {
  constructor(private http: HttpClient) {}

  requestCanalOnboardingSimpleList(): Observable<CanalOnboardingDto[]> {
    const params: HttpParams = new HttpParams();
    const url = environment.api + environment.backendV2.canalOnboardingApi;
    return this.http
      .get<ApiResponse<CanalOnboardingDto[]>>(url, { params })
      .pipe(map((resp) => mapResponseApi<CanalOnboardingDto[]>(url, resp)));
  }
}
